### ISSUE TITLE
[Feature Request] Improve documentation about regex validation rules

### ISSUE THREAD

**Issue by StevenMMortimer** [pyeve/cerberus#547]
Used Cerberus version / latest commit: 1.3.2

- [ ] I have the capacity to improve the docs when my problem is solved.
- [ ] I have the capacity to submit a patch when a bug is identified.
- [x] My question does not concern a practical use-case that I can't figure out
      to solve.

----

#### Use-case abstract

There are a number of references to regex used within validations. For example: 
 - [keysrules](https://docs.python-cerberus.org/en/stable/validation-rules.html?highlight=regex#keysrules)
 - [meta](https://docs.python-cerberus.org/en/stable/validation-rules.html?highlight=regex#meta)
 - [regex](https://docs.python-cerberus.org/en/stable/validation-rules.html?highlight=regex#regex)

And it is not clear to the user what is going to happen to the supplied regex in each validation, which I believe is handled differently across these examples. This is highlighted in the following question and response on SO: https://stackoverflow.com/questions/61734235/using-cerberus-regex-to-validate-string-ends-with-pattern

----

#### Bug report / Feature request

**Okay solution**
Make explicit in documentation where the user-supplied regex is being manipulated within function calls. For example, add to the documentation a statement that says the regex validation rule has to satisfy the entire string because the supplied regex pattern will be wrapped in `^` and `$`.

**Possibly a better solution?**
Alternatively, it would be awesome if the `regex` validation rule was not wrapped at all with the start and end special characters. Then the behavior would match what the documentation implies and give users more flexibility, rather than forcing them to specify a full string match using `.*` in certain places they do not want matched.

**Comment by funkyfuture** [comment]
> And it is not clear to me what is going to happen to the supplied regex in each validation

in the `meta` example it serves to illustrate a consitent use-case. in the `keysrules` example it shows a common use-case to validate all keys in a mapping against a regular expression.

> give users more flexibility, rather than forcing them to specify a full string match using `.*` in certain places they do not want matched.

can you provide an example where a search-in-a-string is preferable to the string-must-match-a-pattern behaviour?

**Comment by StevenMMortimer** [comment]
@funkyfuture Thanks for the additional detail there. I think you answered the SO post referenced above which provides an example of wanting to validate a portion of a string (e.g. ends with `".csv"`) rather than the entire string. 

In that specific case, I imagine there are other ways to validate the file type and/or path represented by the string, but I found it very confusing that supplying `r".csv$"` was not a valid pattern for the regex validation. This is because Cerberus wraps the user supplied pattern with `^` and `$`; however, no where in the documentation does it say that this is what is going to happen to the pattern the user provides. 

As a user, it seems a little bit non-intuitive to be forced into creating a pattern that will validate the entire string instead of just a portion of the string, which is almost always the case when writing regex in the course of writing code. When I use a "regex" validation I expect it to run my regex as I provided it.
